source/Generic/GamesSizeCalculator/GamesSizeCalculator.cs

```diff
@@ -279,16 +295,38 @@ public override void OnLibraryUpdated(OnLibraryUpdatedEventArgs args)
             SavePluginSettings(settings.Settings);
         }
 
+        private ICollection<IOnlineSizeCalculator> GetOnlineSizeCalculators(SteamApiClient steamClient)
+        {
+            var onlineSizeCalculators = new List<IOnlineSizeCalculator>();
+            if (settings.Settings.GetUninstalledGameSizeFromSteam)
+            {
+                onlineSizeCalculators.Add(new SteamSizeCalculator(steamClient, GetDefaultSteamAppUtility()
+                    , settings.Settings.IncludeDlcInSteamCalculation
+                    , settings.Settings.IncludeOptionalInSteamCalculation
+                    , settings.Settings.GetSizeFromSteamNonSteamGames
+                    , settings.Settings.DepotRegionWords.ToArray()
+                    , settings.Settings.DepotRegionWordsBlacklist.ToArray()));
+            }
+            if (settings.Settings.GetUninstalledGameSizeFromGog)
+            {
+                onlineSizeCalculators.Add(new GOG.GogSizeCalculator(new GOG.HttpDownloaderWrapper(), settings.Settings.GetSizeFromGogNonGogGames));
+            }
+            return onlineSizeCalculators;
+        }
+
         private void ProcessGamesOnLibUpdate(GlobalProgressActionArgs a)
         {
-            var games = PlayniteApi.Database.Games.Where(x => x.IsInstalled);
-            a.ProgressMaxValue = games.Count();
+            ICollection<Game> games = PlayniteApi.Database.Games;
+            if (!settings.Settings.GetUninstalledGameSizeFromSteam)
+                games = PlayniteApi.Database.Games.Where(x => x.IsInstalled).ToList();
+
```

Comment:
Use curly braces here please. Perhaps it could be done in one line like this

`ICollection<Game> games = settings.Settings.GetUninstalledGameSizeFromSteam ? PlayniteApi.Database.Games : PlayniteApi.Database.Games.Where(x => x.IsInstalled).ToList()`